packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -225,23 +225,65 @@ export class FilterDescriptor implements vscode.QuickPickItem {
+     * @returns string the credential manager override
+     */
+    public static getCredentialManagerOverride(): string {
         ZoweLogger.trace("ProfilesUtils.getCredentialManagerOverride called.");
-        const knownCredentialManagers = imperative.CredentialManagerOverride.getKnownCredMgrs();
-        const credentialManager = knownCredentialManagers.find((knownCredentialManager) => {
-            try {
-                return vscode.extensions.getExtension(knownCredentialManager.credMgrZEName);
-            } catch (err) {
-                return false;
-            }
-        });
-        if (!credentialManager) {
-            return undefined;
+        const settingsFile = path.join(getZoweDir(), "settings", "imperative.json");
+        const imperativeConfig = JSON.parse(fs.readFileSync(settingsFile).toString());
+        const credentialManagerOverride = imperativeConfig?.overrides[imperative.CredentialManagerOverride.CRED_MGR_SETTING_NAME];
+        if (typeof credentialManagerOverride === "string") {
+            return credentialManagerOverride;
+        }
+        return imperative.CredentialManagerOverride.DEFAULT_CRED_MGR_NAME;
+    }
+
+    /**
+     * Get the map of names associated with the custom credential manager
+     * @param string credentialManager the credential manager display name
+     * @returns imperative.ICredentialManagerNameMap the map with all names related to the credential manager
+     */
+    public static getCredentialManagerMap(credentialManager: string): imperative.ICredentialManagerNameMap | undefined {
+        ZoweLogger.trace("ProfilesUtils.getCredentialManagerNameMap called.");
+        return imperative.CredentialManagerOverride.getCredMgrInfoByDisplayName(credentialManager);
+    }
+
+    /**
+     * Update the current credential manager override
+     * @param setting the credential manager to use in imperative.json
+     */
+    public static async updateCredentialManagerSetting(setting: string): Promise<void> {
+        ZoweLogger.trace("ProfilesUtils.updateCredentialManagerSetting called.");
+        const settingEnabled: boolean = SettingsConfig.getDirectValue(globals.SETTINGS_SECURE_CREDENTIALS_ENABLED);
+        if (settingEnabled) {
+            await globals.setGlobalSecurityValue(setting);
+            imperative.CredentialManagerOverride.recordCredMgrInConfig(setting);
         }
-        return imperative.CredentialManagerOverride.getCredMgrInfoByDisplayName(credentialManager.credMgrDisplayName);
     }
 
-    public static async activateCredentialManagerOverride(
+    /**
+     * Activate a vscode extension of a custom credential manager
+     * @param credentialManagerExtension The credential manager VS Code extension name to activate
+     * @returns Promise<imperative.ICredentialManagerConstructor> the constructor of the activated credential manager
+     */
+    private static async activateCredentialManagerOverride(
```

Comment:
I'm curious if we should keep this methods public to avoid any breakage (if extenders have access to call these).
if not, please disregard this comment 😋

Reply:
It should be fine if this was switched to private, since it's in our Zowe Explorer package and no extenders can access this function :)